dss_metrics/metrics.py

```diff
@@ -64,46 +76,42 @@ def get_metrics(self):
         num_seconds = self.configs['polling_interval_secs']
         num_iterations = 1
 
-        target_proc = None
-        minio_proc = None
-
-        try:
-            target_obj = nvmftarget_collector.NVMFTargetCollector(
-                self.configs, num_seconds, num_iterations,
-                self.whitelist_patterns, filter=self.filter
-            )
-            target_proc = multiprocessing.Process(
-                target=target_obj.poll_statistics,
-                args=[metrics_data_buffer]
-            )
-        except Exception as error:
-            print(f"Unable to instantiate target collector: {str(error)}")
+        collector_procs = []
 
-        try:
-            minio_obj = minio_collector.MinioCollector(
-                self.configs, num_seconds, num_iterations,
-                self.whitelist_patterns, filter=self.filter
-            )
-            minio_proc = multiprocessing.Process(
-                target=minio_obj.poll_statistics,
-                args=[metrics_data_buffer]
-            )
-        except Exception as error:
-            print(f"Unable to instantiate minio collector: {str(error)}")    
+        # initialize collectors
+        target_ustat_obj = nvmftarget_ustat_collector.NVMFTargetUSTATCollector(
+            self.configs, num_seconds, num_iterations, self.whitelist_patterns,
+            filter=self.filter
+        )
+        minio_ustat_obj = minio_ustat_collector.MinioUSTATCollector(
+            self.configs, num_seconds, num_iterations, self.whitelist_patterns,
+            filter=self.filter
+        )
+        minio_rest_obj = minio_rest_collector.MinioRESTCollector(
+            self.configs, self.whitelist_patterns, filter=self.filter
+        )
 
-        try:
-            print("\n launching collector processes.. \n")
-            if target_proc:
-                target_proc.start()
-            if minio_proc:
-                minio_proc.start()
-
-            if target_proc:
-                target_proc.join()
-            if minio_proc:
-                minio_proc.join()
-        except Exception as error:
-            print(f"Failure in collector processes: {str(error)}")
+        # initialize collector processes
+        collector_procs.append(self.create_collector_proc(
+            "target_ustat", target_ustat_obj, metrics_data_buffer))
+        collector_procs.append(self.create_collector_proc(
+            "minio_ustat", minio_ustat_obj, metrics_data_buffer))
+        collector_procs.append(self.create_collector_proc(
+            "minio_rest", minio_rest_obj, metrics_data_buffer))
+
+        # start collectors
+        for proc in collector_procs:
+            try:
+                proc.start()
+            except Exception as error:
+                print(f"Failed to start proccess {proc.name}: {str(error)}")
+
+        # wait for collectors to finish
```

Comment:
Added implementation to have a timeout when waiting for processes to join. If timeout is exceeded process is killed. Since processes share a data buffer for storing metrics, no metrics data is lost that was already processed in this scenario

Reply:
Added Multiprocessing Event Flag that master process sets. Upon setting of this flag, collector processes will attempt to terminate gracefully. 

Afterwards, the timeout option for the join method will add as a secondary check to make sure the collectors are still not running, if they are, they will be terminated.